File: src/libs/parsers/LineParser.ts
import { Figure } from "../figures/Figure";
import { Line } from "../figures/Line";
import { Point } from "../figures/Point";

class LineParser {

    public getData(json:any):Line {
        let line:Line;
        try {
            line = new Line(new Point(json.startX, json.startY), new Point(json.endX, json.endY), json.lineWidth, json.lineColor)
        } catch(err) {
            line = null;
        }
        return line;
    }


}

export { LineParser };

File: src/libs/parsers/LineParser.java
package org.ambrozy.figurepainter.parsers;

import org.ambrozy.figurepainter.figures.Figure;
import org.ambrozy.figurepainter.figures.Line;
import org.ambrozy.figurepainter.figures.Point;

import java.util.Scanner;

public class LineParser {

    private final Scanner scanner;

    public LineParser(Scanner s2) {
        this.scanner = s2;
    }

    public Figure getData() {
        Line line;
        try {
            Point start = new Point(Double.parseDouble(scanner.next()), Double.parseDouble(scanner.next()));
            Point end = new Point(Double.parseDouble(scanner.next()), Double.parseDouble(scanner.next()));
            System.out.println(start.getX() + " " + start.getY() + " " + end.getX() + " " + end.getY());
            line = new Line(start, end);
        } catch(Exception e) {
            return null;
        }

        return line;
    }


}
